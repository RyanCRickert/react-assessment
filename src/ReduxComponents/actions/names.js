export const changeNames = (names = "") => ({
  type: "SET_NAMES",
  names
})