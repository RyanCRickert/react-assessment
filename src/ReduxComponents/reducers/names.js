const namesReducerDefaultState = {
  names: ""
}

export default (state = namesReducerDefaultState, action) => {
  switch(action.type) {
    case "SET_NAMES":
      return {
        ...state,
        names: action.names
      }
    default:
      return state;
  }
}