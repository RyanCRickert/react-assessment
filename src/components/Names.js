import React from 'react';
import { AppConsumer } from "./AppProvider";

export default class Names extends React.Component {
  render() {
    return (
      <AppConsumer>
        {(context) => (
          <textarea
          onChange={context.onChangeText}
          value={context.state.names}
          className="App-name_input"
          />
        )}
    </AppConsumer>
    );
  }
}