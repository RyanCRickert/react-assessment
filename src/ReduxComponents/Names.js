import React from 'react';
import { connect } from "react-redux";
import { changeNames } from "./actions/names";

export class Names extends React.Component {
  onNamesChange = (e) => {
    this.props.changeNames(e.target.value);
  }

  render() {
    return (
      <textarea
      onChange={this.onNamesChange}
      value={this.props.names.names}
      className="App-name_input"
      />
     )
  }
}

const mapStateToProps = (state) => ({
  names: state.names
});

const mapDispatchToProps = (dispatch) => ({
  changeNames: (names) => dispatch(changeNames(names))
})

export default connect(mapStateToProps, mapDispatchToProps)(Names)