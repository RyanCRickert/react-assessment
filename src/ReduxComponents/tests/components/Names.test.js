import React from "react";
import { shallow } from "enzyme";
import { Names } from "../../Names";
import names from "../fixtures/names";

test("should render Names", () => {
  const wrapper = shallow(<Names />)
  expect(wrapper).toMatchSnapshot();
});