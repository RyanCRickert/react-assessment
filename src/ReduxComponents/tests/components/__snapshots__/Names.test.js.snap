// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Names 1`] = `
<textarea
  className="App-name_input"
  onChange={[Function]}
/>
`;
